src/components/Layout/NavBar.component.jsx

```diff
@@ -0,0 +1,111 @@
+import {
+  AppBar,
+  IconButton,
+  fade,
+  makeStyles,
+  Toolbar,
+  Typography,
+} from '@material-ui/core';
+import MenuIcon from '@material-ui/icons/Menu';
+import SearchIcon from '@material-ui/icons/Search';
+import InputBase from '@material-ui/core/InputBase';
+import AccountCircle from '@material-ui/icons/AccountCircle';
+import Switch from '@material-ui/core/Switch';
+import React from 'react';
+
+const useStyles = makeStyles((theme) => ({
+  offset: theme.mixins.toolbar,
+  menuButton: {
+    marginRight: theme.spacing(2),
+  },
+  search: {
+    position: 'relative',
+    borderRadius: theme.shape.borderRadius,
+    backgroundColor: fade(theme.palette.common.white, 0.15),
+    '&:hover': {
+      backgroundColor: fade(theme.palette.common.white, 0.25),
+    },
+    marginRight: theme.spacing(2),
+    marginLeft: 0,
+    width: '100%',
+    [theme.breakpoints.up('sm')]: {
+      marginLeft: theme.spacing(3),
+      width: 'auto',
+    },
+  },
+  searchIcon: {
+    padding: theme.spacing(0, 2),
+    height: '100%',
+    position: 'absolute',
+    pointerEvents: 'none',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  inputRoot: {
+    color: 'inherit',
+  },
+  inputInput: {
+    padding: theme.spacing(1, 1, 1, 0),
+    // vertical padding + font size from searchIcon
+    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
+    transition: theme.transitions.create('width'),
+    width: '100%',
+    [theme.breakpoints.up('md')]: {
+      width: '20ch',
+    },
+  },
+  title: {
+    flexGrow: 1,
+  },
+}));
+
+const Navbar = () => {
+  const classes = useStyles();
+  const [state, setState] = React.useState({
+    checked: false,
+  });
+
+  const handleChange = (event) => {
+    setState({ ...state, [event.target.name]: event.target.checked });
```

Comment:
Not sure if you are adding this for future use cases, but for now, it would be easier to set the hook as:

```javascript
const [isChecked, setIsChecked] = useState(false);

// and then
const handleChange = ({ target }) => {
   setIsChecked(target.checked);
}
```
It is easier to read and use.